packages/lib/src/filterChildren/filterChildren.tsx

```diff
@@ -0,0 +1,94 @@
+import { Children, Fragment, isValidElement, ReactNode } from 'react';
+
+/**
+ * Check if a component has any of the specified static properties.
+ * Handles regular components, standard styled() components, and exotic forward ref styled components.
+ */
+function hasAnyStaticProperty(
+  component: any,
```

Comment:
this should be more strictly typed. Should it be the full `React.Component`, or just the `component.types` parameter